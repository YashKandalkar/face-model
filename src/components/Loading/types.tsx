interface Props {
  primary?: string;
  secondary?: string;
  speed?: number;
}

export const Circular = ({ primary, secondary, speed = 1 }: Props) => {
  return (
    <svg viewBox="0 0 16 16" fill="none" width="50" height="50">
      <circle
        cx="8"
        cy="8"
        r="7"
        stroke={secondary}
        strokeWidth="2"
        vectorEffect="non-scaling-stroke"
      />
      <path
        d="M15 8a7.002 7.002 0 00-7-7"
        stroke={primary}
        strokeWidth="2"
        strokeLinecap="round"
        vectorEffect="non-scaling-stroke"
      >
        <animateTransform
          attributeName="transform"
          type="rotate"
          from="0 8 8"
          to="360 8 8"
          dur={`${speed}s`}
          repeatCount="indefinite"
        />
      </path>
    </svg>
  );
};

export const Linear = ({ primary, secondary, speed = 2 }: Props) => {
  return (
    <div>
      <div />
      <style jsx>{`
        div {
          background: ${secondary};
          border-radius: 2px;
          height: 4px;
          overflow: hidden;
          position: relative;
          width: 100%;
        }
        div > div {
          animation: shift ${speed}s infinite;
          background: ${primary};
          border-radius: 2px;
          height: 4px;
          position: absolute;
        }
        @keyframes shift {
          from {
            left: -80%;
            width: 80%;
          }
          to {
            left: 110%;
            width: 10%;
          }
        }
      `}</style>
    </div>
  );
};
