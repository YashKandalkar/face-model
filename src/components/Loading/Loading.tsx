import { Circular, Linear } from "./types";

interface Props {
  type?: "circular" | "linear" | string;
  primary?: string;
  secondary?: string;
  speed?: number;
}

const Loading = ({
  type,
  primary = "#037bee",
  secondary = "#272727",
  speed,
}: Props) => {
  return type === "circular" ? (
    <Circular primary={primary} secondary={secondary} speed={speed} />
  ) : (
    <Linear primary={primary} secondary={secondary} speed={speed} />
  );
};

export default Loading;
