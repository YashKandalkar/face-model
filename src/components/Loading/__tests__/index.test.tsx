import React from "react";
import { mount } from "enzyme";
import Loading from "../index";

describe("Loading", () => {
  it("should be defined", () => {
    expect(Loading).toBeDefined();
  });

  it("should be rendered", () => {
    const wrapper = mount(<Loading />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render different types", () => {
    const wrapper = mount(
      <div>
        <Loading type="circular" />
        <Loading type="linear" />
      </div>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should match snapshot", () => {
    expect(<Loading type="circular" />).toMatchSnapshot();
    expect(<Loading type="linear" />).toMatchSnapshot();
  });
});
