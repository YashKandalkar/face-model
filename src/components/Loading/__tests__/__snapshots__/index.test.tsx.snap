// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match snapshot 1`] = `
<Loading
  type="circular"
/>
`;

exports[`Loading should match snapshot 2`] = `
<Loading
  type="linear"
/>
`;
